web/static/js/components/folders/FolderShow.jsx

```diff
@@ -104,25 +121,50 @@ class FolderShow extends Component<any, any> {
 
     const readOnly = !project || project.readOnly
 
-    let addButton = null
+    let primaryButton = null
     if (!readOnly) {
-      addButton = (
-        <AddButton text={t('Add survey')} onClick={() => this.newSurvey()} />
-      )
+      if (panelSurvey) {
+        primaryButton = (
+          <RepeatButton text={t('Repeat survey')} disabled={panelSurvey.isRepeatable} onClick={() => this.repeatSurvey()} />
+        )
+      } else {
+        primaryButton = (
+          <AddButton text={t('Add survey')} onClick={() => this.newSurvey()} />
+        )
+      }
+    }
```

Comment:
what do you think if we extract this `primaryButton` to a separate function to avoid scripting in the middle of the `render()` function

Also, with this the `primaryButton` could be a `const` instead of a `let` 😏